SLING-10290 Every request renews sling.formauth token

enapps-enorman opened the pull request:
The problem with the cookie refresh logic is that FormAuthentiactionHandler#getCookieAuthData is looking for the authData in a different place than where FormAuthenticationHandler#createAuthInfo has stored it when jaasHelper is enabled.

enapps-enorman committed:
SLING-10290 Every request renews sling.formauth token

rombert reviewed (commented):
Overall LGTM, thanks @enapps-enorman . A couple of questions:

- I _think_ we could use `JcrResourceConstants.AUTHENTICATION_INFO_CREDENTIALS` without taking an extra dependency since the constant values should be included. If not, we should at least extract it to a constant as we're using the value twice now
- would it be possible to add a test for this?

enapps-enorman committed:
SLING-10290 Every request renews sling.formauth token

Added integration tests to verify the fix

enapps-enorman committed:
SLING-10290 Every request renews sling.formauth token

Use JcrResourceConstants.AUTHENTICATION_INFO_CREDENTIALS constant

enapps-enorman committed:
SLING-10290 Every request renews sling.formauth token

resolve code smell

enapps-enorman commented:
> * I _think_ we could use `JcrResourceConstants.AUTHENTICATION_INFO_CREDENTIALS` without taking an extra dependency since the constant values should be included. If not, we should at least extract it to a constant as we're using the value twice now

Yes, I have changed it to use that constant and the binary looks ok to me.

> * would it be possible to add a test for this?

I've added some simple paxexam integration tests to verify that the http client gets the cookies sent at the appropriate times.  Let me know if you have any concerns with the tests or had something else in mind.

rombert commented:
Thanks @enapps-enorman . Tests looks good, probably a lot of effort to set up, considering the harness we already have in the launchpad-testing module, but very good to have at any rate.
